packages/rancher-vsphere-cpi/charts/templates/vsphere-cpi-ds.yaml

```diff
@@ -21,13 +21,17 @@ spec:
       labels:
         k8s-app: vsphere-cloud-controller-manager
     spec:
-      nodeSelector:
       {{- if .Values.cloudControllerManager.nodeSelector }}
+      nodeSelector:
       {{- with .Values.cloudControllerManager.nodeSelector }}
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- else }}
+      {{- else if eq .Values.clusterType "rke" }}
+      nodeSelector:
         node-role.kubernetes.io/controlplane: "true"
+      {{- else if eq .Values.clusterType "rke2" }}
+      nodeSelector:
+        node-role.kubernetes.io/control-plane: "true"
```

Comment:
This looks good to me, but I wonder if it would be better to specify this via affinityRules rather than nodeSelectors. 

affinityRules would allow us to select a node if one or the other is true, which means we don't have to look for the clusterType at all. (see https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity).

Reply:
Thats interesting question. Can you or @PennyScissors ask RKE2 team about that. Brad may know good reason.

Reply:
I agree this approach would be better, but I think this is something we should look into later on after 2.6 to avoid more changes than necessary.